### PR TITLE
dep: update libxml2 to 2.13.0 and libxslt to 1.1.40

### PR DESCRIPTION
**Pull request by flavorjones**
**What problem is this PR intended to solve?**

- https://gitlab.gnome.org/GNOME/libxml2/-/releases/v2.13.0
- https://gitlab.gnome.org/GNOME/libxslt/-/releases/v1.1.40

### CODE REVIEW

**Comment by flavorjones** [comment]
Looks like we need to wait for a companion libxslt release with gnome/libxslt@75967fc0 to address the compilation issues.

**Comment by flavorjones** [comment]
Added libxslt upgrade to 1.1.40 to this PR

**Comment by flavorjones** [comment]
@rgrove Take note that `sanitize` is going to have some test failures once this makes it into a release (likely v1.17.0 later in the summer):

https://github.com/sparklemotion/nokogiri/actions/runs/9487728125/job/26145008195?pr=3230

```
  1) Failure:
Sanitize::Transformers::CleanDoctype::when :allow_doctype is true#test_0002_should not allow obviously invalid doctype declarations in documents [/__w/nokogiri/nokogiri/sanitize/test/test_clean_doctype.rb:47]:
--- expected
+++ actual
@@ -1 +1 @@
-"<!DOCTYPE html><html>foo</html>"
+"<!DOCTYPE blah><html>foo</html>"
```

**Comment by rgrove** [comment]
> @rgrove Take note that `sanitize` is going to have some test failures once this makes it into a release (likely v1.17.0 later in the summer)

Thanks for the heads up! I'll look into this.

**Comment by flavorjones** [comment]
@rgrove Hmm, I'm digging in and something weird is going on, it's not Sanitize.

**Comment by flavorjones** [comment]
@rgrove Ahhhh ... gnome/libxml2@a3713f78

> Disallow xmlNodeSetName on DTD nodes. DTD nodes don't store the name in a dictionary. Calling xmlNodeSetName with a DTD node could result in an invalid free.

**Comment by rgrove** [comment]
Fun! So it looks like Sanitize may need a small change after all. 🙂

**Comment by flavorjones** [comment]
The [downstream Mechanize test failure](https://github.com/sparklemotion/nokogiri/actions/runs/9487728125/job/26145009739?pr=3230) is due to gnome/libxml2@8d0aaf4b95eaaaaadb4d793b8a3cb05b124ad1a2

Not totally sure what the relationship is between the change and the behavior, but I'm also not totally sure I care, we've seen edge case encoding changes like this before and the Mechanize test in question is more descriptive than prescriptive ... sigh. I'll probably just change the test.

**Comment by flavorjones** [comment]
Ah, this is because Mechanize's `encoding_error?` method is checking the error message contents ... blurgh. Will update that and I think it should be fine.

**Comment by flavorjones** [comment]
See https://github.com/sparklemotion/mechanize/pull/644

**Comment by flavorjones** [comment]
I've opened https://github.com/rgrove/sanitize/pull/238 on sanitize as a potential workaround for the DTD name-change limitation.

**Comment by flavorjones** [comment]
The failure showing up on freebsd is because my upstream fix from https://github.com/sparklemotion/nokogiri/issues/1941 shipped in v2.13.0! Removing the patch in `patches/libxml2/0003-libxml2.la-is-in-top_builddir.patch`! :rocket:
